### ISSUE TITLE
Enhancement: Stop automatic workbench allowing items it doesn't need

### ISSUE THREAD

**Issue by rapus95** [BuildCraft/BuildCraft#3334]
Why the automatic workbench allows stuff in its inventory that it doesn't need? That somewhat doesn't make sense though disallowing it would allow you to remove the diamond pipes in front of it (that only secured that the internal inventory doesn't fill up with crap)

**Comment by rapus95** [comment]
At least for automated insertion from pipes this would be nice

**Comment by LuigiHutch** [comment]
Erm... I just tested this with BC 7.2.6 for 1.8.9 and BC 7.1.16 for 1.7.10.

Items that are not used in the auto workbench recipe are **not** able to enter via automation. I tested with hoppers and pipes where neither were able to put items that are not in the recipe into the auto-workbench. This is one of the setups I used:
![2016-05-12_15 30 32](https://cloud.githubusercontent.com/assets/9326429/15218207/ba751f32-1856-11e6-8114-b5e72f40b661.png)
